src/libs/actions/IOU.js

```diff
@@ -17,94 +17,99 @@ function getPreferredCurrency() {
         })
         .catch(() => Onyx.merge(ONYXKEYS.IOU, {loading: false, creatingIOUTransaction: false, error: true}))
         .finally(() => Onyx.merge(ONYXKEYS.IOU, {loading: false, creatingIOUTransaction: false}));
 }
 
 /**
  * Creates IOUSplit Transaction
- * @param {Object} parameters
- * @param {String} parameters.amount
- * @param {String} parameters.comment
- * @param {String} parameters.currency
- * @param {String} parameters.debtorEmail
+ * @param {Object} params
+ * @param {String} params.amount
+ * @param {String} params.comment
+ * @param {String} params.currency
+ * @param {String} params.debtorEmail
  */
-function createIOUTransaction({
-    comment, amount, currency, debtorEmail,
-}) {
+function createIOUTransaction(params) {
     Onyx.merge(ONYXKEYS.IOU, {loading: true, creatingIOUTransaction: true, error: false});
-    API.CreateIOUTransaction({
-        comment,
-        amount,
-        currency,
-        debtorEmail,
-    })
-        .then(data => data.reportID)
-        .then(reportID => getIOUReportsForNewTransaction([reportID]));
+    API.CreateIOUTransaction(params)
+        .then(data => getIOUReportsForNewTransaction([data]));
 }
 
 /**
  * Creates IOUSplit Transaction
- * @param {Object} parameters
- * @param {Array} parameters.splits
- * @param {String} parameters.comment
- * @param {String} parameters.amount
- * @param {String} parameters.currency
+ * @param {Object} params
+ * @param {Array} params.splits
+ * @param {String} params.comment
+ * @param {String} params.amount
+ * @param {String} params.currency
  */
-function createIOUSplit({
-    comment,
-    amount,
-    currency,
-    splits,
-}) {
+function createIOUSplit(params) {
     Onyx.merge(ONYXKEYS.IOU, {loading: true, creatingIOUTransaction: true, error: false});
 
     API.CreateChatReport({
-        emailList: splits.map(participant => participant.email).join(','),
+        emailList: params.splits.map(participant => participant.email).join(','),
     })
-        .then((data) => {
-            console.debug(data);
-            return data.reportID;
-        })
-        .then(reportID => API.CreateIOUSplit({
-            splits: JSON.stringify(splits),
-            currency,
-            amount,
-            comment,
-            reportID,
+        .then(data => API.CreateIOUSplit({
+            ...params,
+            splits: JSON.stringify(params.splits),
+            reportID: data.reportID,
         }))
-        .then(data => data.reportIDList)
-        .then(reportIDList => getIOUReportsForNewTransaction(reportIDList));
+        .then((data) => {
+            // This data needs to go from this:
+            // {reportIDList: [1, 2], chatReportIDList: [3, 4]}
+            // to this:
+            // [{reportID: 1, chatReportID: 3}, {reportID: 2, chatReportID: 4}]
+            // in order for getIOUReportsForNewTransaction to know which IOU reports are associated with which
+            // chat reports
```

Comment:
I still like the idea of providing a better data structure in Auth, but let's not block the issue. We can create this as a [non-blocking IOU task here](https://github.com/Expensify/Expensify/issues/148974).